doc/regexp/unicode_properties.rdoc

```diff
@@ -0,0 +1,590 @@
+- <tt>/\p{In_Mro}/</tt>
+- <tt>/\p{In_Tangsa}/</tt>
+- <tt>/\p{In_Bassa_Vah}/</tt>
+- <tt>/\p{In_Pahawh_Hmong}/</tt>
+- <tt>/\p{In_Medefaidrin}/</tt>
+- <tt>/\p{In_Miao}/</tt>
+- <tt>/\p{In_Ideographic_Symbols_and_Punctuation}/</tt>
+- <tt>/\p{In_Tangut}/</tt>
+- <tt>/\p{In_Tangut_Components}/</tt>
+- <tt>/\p{In_Khitan_Small_Script}/</tt>
+- <tt>/\p{In_Tangut_Supplement}/</tt>
+- <tt>/\p{In_Kana_Extended_B}/</tt>
+- <tt>/\p{In_Kana_Supplement}/</tt>
+- <tt>/\p{In_Kana_Extended_A}/</tt>
+- <tt>/\p{In_Small_Kana_Extension}/</tt>
+- <tt>/\p{In_Nushu}/</tt>
+- <tt>/\p{In_Duployan}/</tt>
+- <tt>/\p{In_Shorthand_Format_Controls}/</tt>
+- <tt>/\p{In_Znamenny_Musical_Notation}/</tt>
+- <tt>/\p{In_Byzantine_Musical_Symbols}/</tt>
+- <tt>/\p{In_Musical_Symbols}/</tt>
+- <tt>/\p{In_Ancient_Greek_Musical_Notation}/</tt>
+- <tt>/\p{In_Kaktovik_Numerals}/</tt>
+- <tt>/\p{In_Mayan_Numerals}/</tt>
+- <tt>/\p{In_Tai_Xuan_Jing_Symbols}/</tt>
+- <tt>/\p{In_Counting_Rod_Numerals}/</tt>
+- <tt>/\p{In_Mathematical_Alphanumeric_Symbols}/</tt>
+- <tt>/\p{In_Sutton_SignWriting}/</tt>
+- <tt>/\p{In_Latin_Extended_G}/</tt>
+- <tt>/\p{In_Glagolitic_Supplement}/</tt>
+- <tt>/\p{In_Cyrillic_Extended_D}/</tt>
+- <tt>/\p{In_Nyiakeng_Puachue_Hmong}/</tt>
+- <tt>/\p{In_Toto}/</tt>
+- <tt>/\p{In_Wancho}/</tt>
+- <tt>/\p{In_Nag_Mundari}/</tt>
+- <tt>/\p{In_Ethiopic_Extended_B}/</tt>
+- <tt>/\p{In_Mende_Kikakui}/</tt>
+- <tt>/\p{In_Adlam}/</tt>
+- <tt>/\p{In_Indic_Siyaq_Numbers}/</tt>
+- <tt>/\p{In_Ottoman_Siyaq_Numbers}/</tt>
+- <tt>/\p{In_Arabic_Mathematical_Alphabetic_Symbols}/</tt>
+- <tt>/\p{In_Mahjong_Tiles}/</tt>
+- <tt>/\p{In_Domino_Tiles}/</tt>
+- <tt>/\p{In_Playing_Cards}/</tt>
+- <tt>/\p{In_Enclosed_Alphanumeric_Supplement}/</tt>
+- <tt>/\p{In_Enclosed_Ideographic_Supplement}/</tt>
+- <tt>/\p{In_Miscellaneous_Symbols_and_Pictographs}/</tt>
+- <tt>/\p{In_Emoticons}/</tt>
+- <tt>/\p{In_Ornamental_Dingbats}/</tt>
+- <tt>/\p{In_Transport_and_Map_Symbols}/</tt>
+- <tt>/\p{In_Alchemical_Symbols}/</tt>
+- <tt>/\p{In_Geometric_Shapes_Extended}/</tt>
+- <tt>/\p{In_Supplemental_Arrows_C}/</tt>
+- <tt>/\p{In_Supplemental_Symbols_and_Pictographs}/</tt>
+- <tt>/\p{In_Chess_Symbols}/</tt>
+- <tt>/\p{In_Symbols_and_Pictographs_Extended_A}/</tt>
+- <tt>/\p{In_Symbols_for_Legacy_Computing}/</tt>
+- <tt>/\p{In_CJK_Unified_Ideographs_Extension_B}/</tt>
+- <tt>/\p{In_CJK_Unified_Ideographs_Extension_C}/</tt>
+- <tt>/\p{In_CJK_Unified_Ideographs_Extension_D}/</tt>
+- <tt>/\p{In_CJK_Unified_Ideographs_Extension_E}/</tt>
+- <tt>/\p{In_CJK_Unified_Ideographs_Extension_F}/</tt>
+- <tt>/\p{In_CJK_Compatibility_Ideographs_Supplement}/</tt>
+- <tt>/\p{In_CJK_Unified_Ideographs_Extension_G}/</tt>
+- <tt>/\p{In_CJK_Unified_Ideographs_Extension_H}/</tt>
+- <tt>/\p{In_Tags}/</tt>
+- <tt>/\p{In_Variation_Selectors_Supplement}/</tt>
+- <tt>/\p{In_Supplementary_Private_Use_Area_A}/</tt>
+- <tt>/\p{In_Supplementary_Private_Use_Area_B}/</tt>
+- <tt>/\p{In_No_Block}/</tt>
```

Comment:
are unicode age properties, e.g. `\p{Age=1.1}`, omitted on purpose?

Reply:
Yes.  I tried all properties.  About 12-15 raised exceptions, mostly the age ones, but one or two others as well.

Reply:
Which error did you get? `invalid character property name`? This would be strange, because age properties are still supported AFAICT. 

```sh
ruby -e "puts RUBY_VERSION, /\p{Age=1.1}/ =~ 'a'"
# => 3.2.0
# => 0
```

[This list](https://github.com/ammar/regexp_parser/blob/7413c52e0f6dca97c1d21e2daaea040c83eb46cb/lib/regexp_parser/syntax/token/unicode_property.rb#L40-L62) shows at which points each age property was added to Ruby. The only property for which support has ever been removed is `\p{newline}` (dropped in Ruby 2.0). I think the removal of `\p{newline}` is actually the only subtractive change ever made to Ruby's Regexp syntax, at least after Ruby 1.8.6.

Reply:
These supported properties are missing from the doc:

```
age=1.1
age=10.0
age=11.0
age=12.0
age=12.1
age=13.0
age=14.0
age=2.0
age=2.1
age=3.0
age=3.1
age=3.2
age=4.0
age=4.1
age=5.0
age=5.1
age=5.2
age=6.0
age=6.1
age=6.2
age=6.3
age=7.0
age=8.0
age=9.0
alphabetic
ascii_hex_digit
control
decimal_number
in_basic_latin
in_braille_patterns
in_deseret
in_meroitic_hieroglyphs
in_ol_chiki
in_old_north_arabian
in_old_south_arabian
in_palmyrene
in_sharada
in_shavian
lowercase
punctuation
uppercase
white_space
```

Reply:
$ ruby --version
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x64-mingw-ucrt]
$ ruby -e "puts RUBY_VERSION, /\p{Age=1.1}/ =~ 'a'"
-e:1: invalid character property name {Age=1.1}: /\p{Age=1.1}/
-e: compile error (SyntaxError)

Reply:
I see that at least some of those you've listed do work in my Ruby version.
I formed my list of candidates from enc/unicode/15.0.0/name2ctype.h.  Is there a better source?

Reply:
> ruby 3.2.2 (2023-03-30 revision https://github.com/ruby/ruby/commit/e51014f9c05aa65cbf203442d37fef7c12390015) [x64-mingw-ucrt]
> $ ruby -e "puts RUBY_VERSION, /\p{Age=1.1}/ =~ 'a'"
> -e:1: invalid character property name {Age=1.1}: /\p{Age=1.1}/

i've run specs involving age properties on many environments, and this is the first time i've seen this error. it is probably specific to mingw. i guess it could be a bug, or the flag `USE_UNICODE_AGE_PROPERTIES` is purposefully not set there for some memory reason...

> I formed my list of candidates from enc/unicode/15.0.0/name2ctype.h. Is there a better source?

i think `name2ctype` is fine, though for optimization reasons it does not list all properties in the same way. properties that happen to match the same codepoints as others are defined as aliases, e.g. `#define CR_In_Basic_Latin CR_ASCII`. i also have a gem, [regexp_property_values](https://github.com/jaynetics/regexp_property_values), that can provide the list (but it basically just reproduces what ruby's `tool/enc-unicode.rb` does, so might not be failsafe against all future changes).

Reply:
Thanks @jaynetics, for your help.

I'm incline to add the missing properties manually, just to keep it simple.  I've retained the order in the .h file, in case it matters.  Any ideas about where to add, or even about the overall ordering?

Reply:
I see that you've used alpha order over at your gem.  Will do the same.  May I take values from your file `values`?

Reply:
I have done so.

Reply:
> May I take values from your file values?

of course!

Reply:
Just FYI, you can check whether a property is supported or not in the source. For Age=1.1, for example, it shows up at https://github.com/ruby/ruby/blob/master/enc/unicode/15.0.0/name2ctype.h#L43101. Please note that you have to lowercase and eliminate spaces (and I think also hyphens,...).

Reply:
For the ordering of values, please have a look at https://github.com/k-takata/Onigmo/blob/master/doc/UnicodeProps.txt. I think that's better than just simply alphabetic order, because that gives the user at least a faint idea as to what all these properties are about.

Reply:
> i've run specs involving age properties on many environments, and this is the first time i've seen this error. it is probably specific to mingw. i guess it could be a bug, or the flag `USE_UNICODE_AGE_PROPERTIES` is purposefully not set there for some memory reason...

The age properties indeed take quite a bit of memory. The inversion lists of the age properties (in name2ctype.h) are 24373 - 37636 = 13263 lines long, which is more than a third of the about 39000 lines overall for the inversion list data. The problem is that these lists repeat mostly the same data over and over again, because what's in version 15 is of course also in version 14 is in version 13... I'll look into how to improve that, but this may take time.

Reply:
> For the ordering of values, please have a look at https://github.com/k-takata/Onigmo/blob/master/doc/UnicodeProps.txt. I think that's better than just simply alphabetic order, because that gives the user at least a faint idea as to what all these properties are about.

@duerst, revised and much better; thanks.

Reply:
IIUC, the parts of Onigmo that are used by Ruby are now maintained within the ruby repository, and `UnicodeProps.txt` from the original repository hasn't been updated since 2019, so it is missing 40 or so properties added since then.